packages/react-native/ReactCommon/react/renderer/components/view/ViewProps.cpp

```diff
@@ -440,28 +440,16 @@ std::array<float, 3> ViewProps::getTranslateForTransformOrigin(float viewWidth,
   float viewCenterY = viewHeight / 2;
 
   std::array<float, 3> origin = {viewCenterX, viewCenterY, 0.0f};
-  std::istringstream iss(transformOrigin);
-  std::string part;
-  for (int i = 0; std::getline(iss, part, ' ') && i < 3; i++) {
-    auto percentPos = part.find('%');
-    bool isPercent = percentPos != std::string::npos;
-    if (isPercent) {
-      origin[i] = (i == 0 ? viewWidth : viewHeight) * std::stof(part.substr(0, percentPos)) / 100.0f;
-    } else if (part == "top") {
-       origin[1] = 0.0f;
-    } else if (part == "bottom") {
-       origin[1] = static_cast<float>(viewHeight);
-    } else if (part == "left") {
-       origin[0] = 0.0f;
-    } else if (part == "right") {
-       origin[0] = static_cast<float>(viewWidth);
-    } else if (part == "center") {
-       continue;
-    } else {
-       origin[i] = std::stof(part);
+  
+  for (size_t i = 0; i < 3; ++i) {
+    if (transformOrigin.origin[i].unit == YGUnitPoint) {
```

Comment:
Could we pull `transformOrigin.origin[i]` into a variable in the loop since it's referenced a lot?

Reply:
Done!